src/server/engine_shard_set.cc

```diff
@@ -54,6 +54,22 @@ constexpr uint64_t kCursorDoneState = 0u;
 
 vector<EngineShardSet::CachedStats> cached_stats;  // initialized in EngineShardSet::Init
 
+struct ShardMemUsage {
+  std::size_t commited = 0;
+  std::size_t used = 0;
+};
+
+std::ostream& operator<<(std::ostream& os, const ShardMemUsage& mem) {
+  return os << "commited: " << mem.commited << " vs used " << mem.used;
+}
+
+ShardMemUsage ReadShardMemUsage() {
+  std::size_t ignore = 0;
+  ShardMemUsage usage;
+  zmalloc_get_allocator_info(&usage.used, &ignore, &usage.commited);
```

Comment:
I do not think this is the best we can do here. If we already go over areas, we should apply our own logic of discarding areas that do not fit  mem_utilization_threshold, 
which btw should be renamed to `mem_defrag_page_utilization_threshold`

Reply:
changed - we are not using a dedicated function that calculating the wasted memory.